### PR TITLE
Parameter scopes computation

### PR DESCRIPTION
**Pull request by litvinovg**
**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues/3662)**

- Created rdf classes (interfaces) in ontology
- Created java interfaces to get data for scopes computation

# Interested parties
Tag (@ mention) interested parties or, if unsure, @VIVO-project/vivo-committers

### CODE REVIEW

File: home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl [hunk 1]
```diff
@@ -399,6 +399,8 @@
     <!-- https://vivoweb.org/ontology/vitro-dynamic-api#action -->
 
     <owl:Class rdf:about="https://vivoweb.org/ontology/vitro-dynamic-api#action">
+        <rdfs:subClassOf rdf:resource="https://vivoweb.org/ontology/vitro-dynamic-api#operation"/>
+		<rdfs:subClassOf rdf:resource="https://vivoweb.org/ontology/vitro-dynamic-api#parameterProvider"/>
```
Comment (chenejac) [inline home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl:403]: Please correct indentation

Comment (chenejac) [reply home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl:403]: Not sure Action shouldn't be parameterReceiver as well. Think also about the case when we are chaining an action in the another action which we discussed once.

File: home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl [hunk 2]
```diff
@@ -472,13 +488,14 @@
...
     <owl:Class rdf:about="https://vivoweb.org/ontology/vitro-dynamic-api#operation">
-        <rdfs:label xml:lang="en-US">Operation</rdfs:label>
+        <rdfs:label xml:lang="en-US">Operation interface</rdfs:label>
     </owl:Class>
 
 
     <!-- https://vivoweb.org/ontology/vitro-dynamic-api#template -->
 
     <owl:Class rdf:about="https://vivoweb.org/ontology/vitro-dynamic-api#template">
+        <rdfs:subClassOf rdf:resource="https://vivoweb.org/ontology/vitro-dynamic-api#parameterReceiver"/>
```
Comment (chenejac) [inline home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl:498]: In your example (https://app.diagrams.net/#G176In5cQ-tWecx0aP0ILBTODuCoiYk-0o) N3Template is parameter receiver and provider?

File: home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl (hunk 1 above)
Comment (chenejac) [reply home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl:403]: I found later in your code, there are requiredParams for actions but it is calculated based on set of steps, not defined in RDF. Just ignore above comment.

**Review by chenejac — CHANGES_REQUESTED** [review]

File: home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl (hunk 2 above)
Comment (litvinovg) [reply home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl:498]: My example has a mistake. It shouldn't be a provider

File: home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl (hunk 1 above)
Comment (litvinovg) [reply home/src/main/resources/rdf/tbox/filegraph/vitro-dynamic-api.owl]: Indentation is fixed

**Review by wwelling — APPROVED** [review]

**Review by chenejac — APPROVED** [review]

**Pull request merged**